cmd/bosun/expr/funcs.go

```diff
@@ -384,6 +384,161 @@ var builtins = map[string]parse.Func{
 		F:       V,
 		MapFunc: true,
 	},
+	"aggregate": {
+		Args:   []models.FuncType{models.TypeSeriesSet, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   tagFirst,
+		F:      Aggregate,
+	},
+}
+
+// Aggregate combines multiple series matching the specified groups using an aggregator function. If group
+// is empty, all given series are combined, regardless of existing groups.
+// Available aggregator functions include: avg (average), p50 (median), min (minimum) and max (maximum).
+func Aggregate(e *State, T miniprofiler.Timer, series *Results, groups string, aggregator string) (*Results, error) {
+	results := Results{}
+
+	grps := splitGroups(groups)
+	if len(grps) == 0 {
+		// no groups specified, so we merge all group values
+		res, err := aggregate(e, T, series, aggregator)
+		if err != nil {
+			return &results, err
+		}
+		res.Group = opentsdb.TagSet{}
+		results.Results = append(results.Results, res)
+	} else {
+		// at least one group specified, so we work out what
+		// the new group values will be
+		newGroups := map[string]*Results{}
+		for _, result := range series.Results {
+			vals := []string{}
+			for _, grp := range grps {
+				if val, ok := result.Group[grp]; ok {
+					vals = append(vals, val)
```

Comment:
No need for else, can `continue in the if statement, and then have the error after the if statement. Same idea of minimal indentaiton

Reply:
Done 👍